apps/marketplace/actions/memberActions.js

```diff
@@ -42,50 +40,59 @@ export const handleSignupSubmit = model => (dispatch, getState) => {
   })
 }
 
-// LOAD USER REGISTRATION DETAILS FOR SIGNUP COMPLETION FORM
-export const handleLoadSignupSuccess = response => ({
-  type: LOAD_SIGNUP_SUCCESS,
+// PROVIDE PASSWORD AND/OR NAME TO COMPLETE ACCOUNT CREATION
+export const handleCreateUserSuccess = response => ({
+  type: CREATE_USER_SUCCESS,
   data: response.data
 })
 
-export const loadSignup = token => dispatch => {
+export const createUser = (token, emailAddress, password) => (dispatch, getState) => {
   dispatch(sendingRequest(true))
-  dmapi({ url: `/signup/validate-invite/${token}` }).then(response => {
+  dmapi({
+    method: 'post',
+    url: `/create-user/${token}`,
+    params: { e: encodeURIComponent(emailAddress) },
+    headers: {
+      'X-CSRFToken': getState().app.csrfToken,
+      'Content-Type': 'application/json'
+    },
+    data: { password }
+  }).then(response => {
     if (response.error) {
-      dispatch(setErrorMessage(REGISTRATION_NOT_FOUND))
+      if (response.status === 409) {
+        dispatch(setErrorMessage(DUPLICATE_USER))
```

Comment:
Doesn't this leak email addresses if tried enough times?

I also thought that the system would send an email about forgotten password when an email match happens.

Reply:
This is part of the create user request, which should now only process with a valid token and email address pair, which should not be created if an email address is already linked to an account - it will do as you said, i.e. say the signup was successful and to look out for an email, but the email sent is the one saying you're already a user. So I don't think this duplicate user check is relevant anymore.

The API will only respond with a 409 if a seller tries to signup using a (non generic) email domain that is already associated with an existing supplier (in which case an existing account needs to use the 'invite contributor' feature).